lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -375,7 +383,12 @@ void THeaderTransport::resetProtocol() {
 }
 
 uint32_t THeaderTransport::getWriteBytes() {
-  return wBase_ - wBuf_.get();
+  ptrdiff_t wb = wBase_ - wBuf_.get();
+  if (wb > std::numeric_limits<uint32_t>().max()) {
+    throw TTransportException(TTransportException::CORRUPTED_DATA,
+                              "write size is unreasonable");
+  }
```

Comment:
I added apache::thrift::transport::safe_numeric_cast to the TTransportException header.  I couldn't think of a better place to put it for now... I agree it simplifies the code in many places, and the transport code seems to be the code closest to the platform where these transform issues exist.